Flaky test: test_status_returns_last_cycle_when_idle - budget tracker race condition

## Summary

The test `test_status_returns_last_cycle_when_idle` in `tests/unit/agent_tests/test_agent_service_new.py` is flaky due to a race condition with the budget tracker's file operations during parallel test execution.

## Error

```
FAILED tests/unit/agent_tests/test_agent_service_new.py::TestAgentServiceGetStatus::test_status_returns_last_cycle_when_idle 
- json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

The error occurs in `ktrdr/agents/budget.py:66` when `_load_today()` tries to load a JSON file that is empty.

## Root Cause

This is a test isolation issue. The budget tracker uses a shared file path, and when tests run in parallel:
1. One test creates/truncates the budget file
2. Another test tries to read it while it's empty
3. `json.loads()` fails on empty string

## Current Mitigation

The test has been marked as `@pytest.mark.skip` to unblock CI.

## Proposed Fix

Options:
1. **Mock the budget tracker** in this test to avoid file operations
2. **Use test-specific file paths** for the budget tracker via fixture
3. **Add file locking** to the budget tracker (overkill for this use case)

Option 1 (mocking) is likely the cleanest approach for unit tests.

## Files Affected

- `tests/unit/agent_tests/test_agent_service_new.py` - the flaky test
- `ktrdr/agents/budget.py` - the budget tracker with file operations

--- a/tests/unit/agent_tests/test_agent_service_new.py
+++ b/tests/unit/agent_tests/test_agent_service_new.py
@@ -138,7 +138,7 @@ class TestAgentServiceTrigger:
                 try:
                     await task
                 except (asyncio.CancelledError, Exception):
-                    pass
+                    pass  # Best-effort cleanup: ignore errors from cancelled tasks
 
     @pytest.fixture(autouse=True)
     def mock_budget(self):
@@ -256,9 +256,15 @@ class TestAgentServiceGetStatus:
         mock_tracker.get_remaining.return_value = 5.0
         mock_tracker.daily_limit = 5.0
 
-        with patch(
-            "ktrdr.api.services.agent_service.get_budget_tracker",
-            return_value=mock_tracker,
+        with (
+            patch(
+                "ktrdr.api.services.agent_service.get_budget_tracker",
+                return_value=mock_tracker,
+            ),
+            patch(
+                "ktrdr.agents.workers.research_worker.get_budget_tracker",
+                return_value=mock_tracker,
+            ),
         ):
             yield mock_tracker
 
@@ -406,9 +412,15 @@ class TestAgentServiceMetadataContract:
         mock_tracker.get_remaining.return_value = 5.0
         mock_tracker.daily_limit = 5.0
 
-        with patch(
-            "ktrdr.api.services.agent_service.get_budget_tracker",
-            return_value=mock_tracker,
+        with (
+            patch(
+                "ktrdr.api.services.agent_service.get_budget_tracker",
+                return_value=mock_tracker,
+            ),
+            patch(
+                "ktrdr.agents.workers.research_worker.get_budget_tracker",
+                return_value=mock_tracker,
+            ),
         ):
             yield mock_tracker
 
@@ -1272,9 +1284,15 @@ class TestAgentServiceMultiResearchStatus:
         mock_tracker.get_remaining.return_value = 5.0
         mock_tracker.daily_limit = 5.0
 
-        with patch(
-            "ktrdr.api.services.agent_service.get_budget_tracker",
-            return_value=mock_tracker,
+        with (
+            patch(
+                "ktrdr.api.services.agent_service.get_budget_tracker",
+                return_value=mock_tracker,
+            ),
+            patch(
+                "ktrdr.agents.workers.research_worker.get_budget_tracker",
+                return_value=mock_tracker,
+            ),
         ):
             yield mock_tracker
 

--- a/tests/unit/agent_tests/test_multi_research.py
+++ b/tests/unit/agent_tests/test_multi_research.py
@@ -642,7 +642,7 @@ class TestTriggerCapacityCheck:
                 try:
                     await task
                 except (asyncio.CancelledError, Exception):
-                    pass
+                    pass  # Best-effort cleanup: ignore errors from cancelled tasks
 
     @pytest.fixture(autouse=True)
     def mock_budget(self):

--- a/tests/unit/agents/test_research_worker_multi.py
+++ b/tests/unit/agents/test_research_worker_multi.py
@@ -578,7 +578,7 @@ class TestCapacityCheck:
                 try:
                     await task
                 except (asyncio.CancelledError, Exception):
-                    pass
+                    pass  # Best-effort cleanup: ignore errors from cancelled tasks
 
     @pytest.fixture(autouse=True)
     def mock_budget(self):
@@ -738,7 +738,7 @@ class TestCoordinatorLifecycle:
                 try:
                     await task
                 except (asyncio.CancelledError, Exception):
-                    pass
+                    pass  # Best-effort cleanup: ignore errors from cancelled tasks
 
     @pytest.fixture(autouse=True)
     def mock_budget(self):

